File: magellan-sample-advanced/src/main/java/com/wealthfront/magellan/sample/advanced/SampleApplication.java
package com.wealthfront.magellan.sample.advanced;

import android.app.Application;
import android.content.Context;

public class SampleApplication extends Application {

  private static AppComponent appComponent;

  @Override
  public void onCreate() {
    super.onCreate();
    appComponent = DaggerAppComponent.builder().appModule(new AppModule()).build();
  }

  public static AppComponent injector() {
    return appComponent;
  }

}

File: magellan-sample-migration/src/main/java/com/wealthfront/magellan/sample/migration/SampleApplication.kt
package com.wealthfront.magellan.sample.migration

import android.app.Application
import android.content.Context

class SampleApplication : Application(), AppComponentContainer {

  private lateinit var appComponent: AppComponent

  override fun onCreate() {
    super.onCreate()
    appComponent = DaggerAppComponent.builder()
      .appModule(AppModule)
      .dogApiModule(DogApiModule)
      .toolbarHelperModule(ToolbarHelperModule)
      .build()
  }

  override fun injector(): AppComponent {
    return appComponent
  }

  companion object {
    @JvmStatic
    fun app(context: Context): AppComponentContainer {
      return context.applicationContext as AppComponentContainer
    }
  }
}
